elftools/dwarf/dwarf_expr.py

```diff
@@ -171,9 +178,33 @@ def parse_arg_struct(arg_struct):
     def parse_arg_struct2(arg1_struct, arg2_struct):
         return lambda stream: [struct_parse(arg1_struct, stream),
                                struct_parse(arg2_struct, stream)]
+ 
+    # ULEB128, then an expression of that length
+    def parse_nestedexpr():
+        def parse(stream):
+            size = struct_parse(structs.Dwarf_uleb128(''), stream)
+            nested_expr_blob = stream.read(size)
+            if isinstance(nested_expr_blob, str): # Python 2 patch. In Py3, it comes as bytes.
+                nested_expr_blob = [ord(b) for b in nested_expr_blob]
+            return [DWARFExprParser(structs).parse_expr(nested_expr_blob)]
+        return parse
+
+    # In Python 2, read returns a str, in Byton 3 - bytes.
+    # For Python 2, bytearray will do, since bytes is a synonym of str and isn't a collection of ints.
+    def to_blob(blob):
```

Comment:
I'm pretty sure you could create a struct in `structs.py` to parse this. There are many examples in `elf/structs.py` and `dwarf/structs.py`. A positive fallout would be that construct is already ported for py2/py3 and you wouldn't need these conditions here.

Generally, in pyelftools all py2/py3 differences are stashed into py3compat or the construct parsing code (except some small things like displaying bytes/string)

Reply:
You do it this way:
```
  [struct_parse(structs.Dwarf_uint8(''), stream) for i in range(len)]
```
Getting an array of ints (which I don't think is a good Pythonic data structure for a blob). But there's no helper function to that effect that I can see. I could create one...